File: python-sandbox/pytest-tests/src/calculator.py
"""Tiny 'system under test' used by the example pytest tests."""

from __future__ import annotations


def add(a: int, b: int) -> int:
    return a + b


def divide(dividend: float, divisor: float) -> float:
    if divisor == 0:
        raise ZeroDivisionError("division by zero")
    return dividend / divisor


def is_even(n: int) -> bool:
    return n % 2 == 0


def fizzbuzz(n: int) -> str:
    if n % 15 == 0:
        return "FizzBuzz"
    if n % 3 == 0:
        return "Fizz"
    if n % 5 == 0:
        return "Buzz"
    return str(n)


class BankAccount:
    """Small stateful class to demonstrate fixtures and class-based tests."""

    def __init__(self, balance: float = 0.0) -> None:
        self.balance = balance

    def deposit(self, amount: float) -> None:
        if amount <= 0:
            raise ValueError("amount must be positive")
        self.balance += amount

    def withdraw(self, amount: float) -> None:
        if amount > self.balance:
            raise ValueError("insufficient funds")
        self.balance -= amount

File: js-sandbox/ts_test/src/calculator.ts
// Tiny typed "system under test" used by the example tests.

export function add(a: number, b: number): number {
  return a + b;
}

export function divide(dividend: number, divisor: number): number {
  if (divisor === 0) {
    throw new Error("division by zero");
  }
  return dividend / divisor;
}

export function isEven(n: number): boolean {
  return n % 2 === 0;
}

export type FizzBuzz = "Fizz" | "Buzz" | "FizzBuzz" | string;

export function fizzbuzz(n: number): FizzBuzz {
  if (n % 15 === 0) return "FizzBuzz";
  if (n % 3 === 0) return "Fizz";
  if (n % 5 === 0) return "Buzz";
  return String(n);
}

export class BankAccount {
  constructor(public balance: number = 0) {}

  deposit(amount: number): void {
    if (amount <= 0) throw new Error("amount must be positive");
    this.balance += amount;
  }

  withdraw(amount: number): void {
    if (amount > this.balance) throw new Error("insufficient funds");
    this.balance -= amount;
  }
}
